accepted/00083-image_management.md

```diff
@@ -0,0 +1,291 @@
+   below.
+
+2. `revision` will be incremented each time an image with the same
+   name and version is built.
+   The image will be identified by `name-version-revision` as in
+   Build Service.
+   Add the revision to image pillar and adjust saltboot and image-sync formulas
+   to handle it.
+   The actual image file will be created before `revision`,
+   so the file name will still contain the build_id in order to make it unique.
+   For details see the Image building workflow below.
+
+
+```
+
+images:
+    POS_Image_Graphical7:  # <name>
+        7.0.0-1:           # <version>-<revision>
+            arch: x86_64
+            basename: POS_Image_Graphical7.x86_64-7.0.0
+            boot_image: POS_Image_Graphical7-7.0.0
+    ...
+            sync:
+    ...
+                bundle_url: https://server.tf.local/os-images/1//POS_Image_Graphical7.x86_64-7.0.0-build33.tar.xz
+
+```
+
+
+3. Do not re-use `suseImageInfo` entries, create new entry for each build.
+   Add yes_no column `obsolete` to `suseImageInfo`.
+   Drop table `suseImageBuildHistory`.
+   Change foreign key of `suseImageRepoDigest` table from `suseImageBuildHistory.id` to `suseImageInfo.id`.
+
+   For Docker the store can hold only one `name-version` image. Pushing new revision
+   to the store replaces the old one. So the old `suseImageInfo` entries can be set to `obsolete` when the
+   new image is built successfully.
+   On failed build the store still contains the previous revision and the corresponding entry
+   in `suseImageInfo` stays unchanged.
+
+   For details see the Image building workflow below.
+
+   For Kiwi it is possible to have multiple revisions of the same `name-version`. So the entries
+   can be deleted individually. Typically an old image can be deleted when the new image is
+   synced to all Branch servers.
+
+   To maintain previous functionality, add functions to 'delete all revisions'
+   and 'delete all obsolete revisions' of an image.
+
+   `suseImageBuildHistory` will need special attention during migration - it will be necessary
+   to create new row in `suseImageInfo` for each row from `suseImageBuildHistory`
+
+4. Add `pillar_id` and `file` to `suseImageInfo`.
+
+   `pillar_id` points to pillar ID in `suseSaltPillars` table
+
+   Note: According to the RFC https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00080-jsonb-pillar-db-storage.md
+   there is a constraint `UNIQUE (org_id, category)` in `suseSaltPillars`.
+   The `category` for image pillars will be set to `"image_" + suseImageInfo.id`
+
+   `file` is a local path for images in SUMA Store or URL for
+   external images (determined by existing `external` flag).
+
+   Pillars and local image files will be deleted when the image is deleted
+   in UI or API.
+
+   `file` will be used to show image URL in UI.
+
+   Pillar entries will be deleted when the corresponding `suseImageInfo` entry is deleted.
+   This will be implemented with an sql trigger.
```

Comment:
1) I am going a little off topic here but in my workshop draft I needed to have uncompressed bundle on the SUMA server so I could register initrd and kernel to cobbler distro. For that I got rid of bundle upload and instead uploaded KIS image individually. For this having single `file` column would not be enough. What do you think about creating new linking table image id <-> file?
2) How about not using `file`, but `URI` instead? As it could be either `file://` or URL in the future?

Reply:
1. good idea, added.
2. I think that using `file://` for local files just complicates things, so `file` is probably better name.